common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java

```diff
@@ -0,0 +1,158 @@
+        final MicrosoftStsTokenResponse msTokenResponse = asMicrosoftStsTokenResponse(response);
+        final MicrosoftStsAccount msAccount = (MicrosoftStsAccount) msStrategy.createAccount(msTokenResponse);
+        final ClientInfo clientInfo = new ClientInfo(msTokenResponse.getClientInfo());
+
+        final Account account = new Account();
+        account.setUniqueId(formatUniqueId(clientInfo));
+        account.setEnvironment(msRequest.getAuthority().toString()); // host of authority with optional port
+        account.setRealm(msAccount.getTenantId()); //tid
+        account.setAuthorityAccountId(tokenClaims.get(OJBECT_ID)); // oid claim from id token
+        account.setUsername(tokenClaims.get(PREFERRED_USERNAME));
+        account.setAuthorityType(AUTHORITY_TYPE);
+        account.setFirstName(tokenClaims.get(GIVEN_NAME));
+        account.setLastName(tokenClaims.get(FAMILY_NAME));
+
+        return account;
+    }
+
+    @NonNull
+    private static MicrosoftStsTokenResponse asMicrosoftStsTokenResponse(final TokenResponse response) {
+        MicrosoftStsTokenResponse msTokenResponse;
+        if (response instanceof MicrosoftStsTokenResponse) {
+            msTokenResponse = (MicrosoftStsTokenResponse) response;
+        } else {
+            throw new IllegalArgumentException("Invalid TokenResponse type.");
+        }
+        return msTokenResponse;
+    }
+
+    @NonNull
+    private static MicrosoftStsAuthorizationRequest asMicrosoftStsAuthorizationRequest(final AuthorizationRequest request) {
+        MicrosoftStsAuthorizationRequest msRequest;
+        if (request instanceof MicrosoftStsAuthorizationRequest) {
+            msRequest = (MicrosoftStsAuthorizationRequest) request;
+        } else {
+            throw new IllegalArgumentException("Invalid AuthorizationRequest type.");
+        }
+        return msRequest;
+    }
+
+    @NonNull
+    private static MicrosoftStsOAuth2Strategy asMicrosoftStsOAuth2Strategy(final OAuth2Strategy strategy) {
+        MicrosoftStsOAuth2Strategy msStrategy;
+        if (strategy instanceof MicrosoftStsOAuth2Strategy) {
+            msStrategy = (MicrosoftStsOAuth2Strategy) strategy;
+        } else {
+            throw new IllegalArgumentException("Invalid strategy type.");
+        }
+        return msStrategy;
+    }
+
+    @Override
+    public AccessToken createAccessToken(
+            final OAuth2Strategy strategy,
+            final AuthorizationRequest request,
+            final TokenResponse response) {
+        final AccessToken accessToken = new AccessToken();
+        accessToken.setTarget(getTarget(request));
+        accessToken.setCachedAt(getCachedAt()); // generated @ client side
+        accessToken.setExpiresOn(getExpiresOn(response)); // derived from expires_in
+        accessToken.setClientInfo(getClientInfo(response));
+        // TODO Do AccessTokens track a family id?
+        //accessToken.setFamilyId(msTokenResponse.getFamilyId());
```

Comment:
@shoatman From the spec, it doesn't look like we're tracking family id in AT cache values. Does this sound approximately correct?